### PR TITLE
Step 2: Prevent user from typing inside of readonly entities around the Readonly Inline Entity

### PR DESCRIPTION
**Pull request by BryanValverdeU**
Prevent user typing inside of Delimiters so we can handle events such as Delete and Arrow Keys cursor movement in next steps.
When user types inside of a delimiter, move the cursor outside of the Delimiter, either before of after depending of the type of delimiter. 
Delimiter

### CODE REVIEW

File: packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts [hunk 1]
```diff
@@ -24,11 +38,68 @@ export function inlineEntityOnPluginEvent(event: PluginEvent) {
...
                     if (isReadOnly(entity)) {
-                        removeDelimiters(wrapper);
+                        removeDelimiters(entity.wrapper);
                     }
                     break;
             }
             break;
+
+        case PluginEventType.KeyDown:
+        case PluginEventType.KeyUp:
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts:48]: I think just handling KeyDown event should be enough

File: packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts [hunk 2]
```diff
@@ -24,11 +38,68 @@ export function inlineEntityOnPluginEvent(event: PluginEvent) {
...
+            if (isCharacterValue(event.rawEvent)) {
+                const position = editor.getFocusedPosition();
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts:50]: What do we want to do if selection is not collapsed?

getFocusedPosition() will always give you a single position event selection is collapsed.

And what do we want to do for table selection and image selection?

**Comment by JiuqingSong** [comment]
Is it possible that cursor is just BETWEEN delimiter and entity, but not inside delimiter?

File: packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts (hunk 2 above)
Comment (BryanValverdeU) [reply packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts:50]: Good point, in this case we should only handle it when the selection is collapsed.

**Comment by BryanValverdeU** [comment]
> Is it possible that cursor is just BETWEEN delimiter and entity, but not inside delimiter?

Right now yes, using arrow keys. in the next step I am going to add a content edit feature to prevent this from happenning.

File: packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts (hunk 1 above)
Comment (BryanValverdeU) [reply packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts]: Removed KeyUp as suggested

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**